### DISCUSSION TITLE
how do you maintain a list/collection of favorite packages and settings?

### DISCUSSION THREAD

**Discussion by pandrewhk** [Homebrew/discussions#4097] [Everyday usage] [answered]
### Output of `brew config`

```shell
-
```

### Output of `brew doctor`

```shell
-
```

### Description of issue

Couldn't find an answer with Google or here on Github yet.

What are best practices for maintaining a list of favorite packages and services / settings to recover your brew-managed environment on a new [macOS] system? Could be part of the backup+restore flow or just getting a new system up to your preferred state.

Ideally mostly automatic, would track the state of everything brew on your current system and can recover mostly the same state elsewhere. 

I can come up with multiple ways to do it, just asking if there are recommended practices already.

**Comment by SMillerDev** [comment, marked as answer]
https://github.com/Homebrew/homebrew-bundle

**Reply by miguelmato** [reply]
does `brew bundle dump` transfer the apps' preferences?

I think that brew should have a way to transfer casks preferences easily. For now I'm using [mackup](https://github.com/lra/mackup), but I would love if brew integrated this functionality as the mackup repo is not up-to-date: for example, [Mos](https://mos.caldis.me/)' preference plist [is missing in mackup](https://github.com/lra/mackup/blob/master/mackup/applications/mos.cfg) but [it's in the brew cask repo ](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/mos.rb).

I'm a newbie in FOSS and coding, but I would love adding this into brew if people are interested.

**Reply by SMillerDev** [reply]
It does not, but feel free to make a pull request. I'd be happy to help

**Reply by tklaver** [reply]
I really wish there was a way to "move" settings to a new setup (in a reliable way).

When you uninstall an app via e.g. [AppCleaner](https://freemacsoft.net/appcleaner/), it lists all of the attached files as well. The technology is there -- I just want to save/backup those settings automatically!
